core/src/main/scala/cromwell/core/simpleton/WomValueBuilder.scala

```diff
@@ -109,6 +111,27 @@ object WomValueBuilder {
       case pairType: WomPairType =>
         val groupedByLeftOrRight: Map[PairLeftOrRight, Traversable[SimpletonComponent]] = group(components map descendIntoPair)
         WomPair(toWomValue(pairType.leftType, groupedByLeftOrRight(PairLeft)), toWomValue(pairType.rightType, groupedByLeftOrRight(PairRight)))
+      case WomObjectType =>
+        val groupedByMapKey: Map[String, Traversable[SimpletonComponent]] = group(components map descendIntoMap)
+        // map keys are guaranteed by the WDL spec to be primitives, so the "coerceRawValue(..).get" is safe.
+        val map: Map[String, WomValue] = groupedByMapKey map { case (k, ss) => k -> toWomValue(WomAnyType, ss) }
+        WomObject(map)
+      case WomAnyType =>
+        // Ok, we're going to have to guess, but the keys should give us some clues:
+        if (components forall { component => MapElementPattern.findFirstMatchIn(component.path).isDefined }) {
+          // Looks like an object
+          toWomValue(WomObjectType, components)
```

Comment:
Could it not be a `Map` ?

Reply:
It could, but to check that I'd need to analyze every entry to make sure the value types are always the same. This just seemed easier (and I believe they're coerceable with a run-time check)

Structs should make this oh so much easier since we won't have to guess anymore

Reply:
I added a test case to make sure we can coerce the result back into the original `Map` type